### DISCUSSION TITLE
DIY battery to Growatt high voltage (MOD 8000TL3-XH)

### DISCUSSION THREAD

**Discussion by tmlarsson** [ai-republic/bms-to-inverter#30] [Q&A] [answered]
Morning. I found this great project yesterday and started to wonder a bit about how the integration between a battery and an inverter actually works.

A bit of background. I have MOD 8000TL3-XH inverter but am reluctant to buy growatts battery as i recon 3000EUR (including swedish 50% grant) for 10kWh battery is quite expensive. Hence, I have been thinking about getting a victron system as I would be able to install 3 multiplus 3000 and 10kWh diy lifepo4 battery for the same amount (roughly).

However, in the meantime, getting a diy battery integrated into the growatt inverter would be awesome. I get that you can get the comminication between a bms and the inverter, but how would you make sure that the voltage input is in the accepted range?

As I understand, the ARK XH BMS block includes a dc-dc step up converter from  ~48v to 400v. I would assume that this is still needed even if you managed to use your own 48v battery? Can we hack the ark xh bms block to accept e.g. a daly bms? That would be quite nice indeed.

**Comment by ai-republic** [comment, marked as answer]
Hi, it should be possible but you would need something hooked up after the BMSes which usually handle the 48V to step up the DC voltage to 400V. You'll need to research that :)

**Comment by LaurisK** [comment]
Hello everyone,
Are you sure about 48V DC - DC? To me it seems that this family uses 600-950V ARK or APX battery packs. since this topic is a bit interesting for me too. As I would like to connect self made battery pack to Growatt `MOD 10KTL3-XH BP` with not growat BMS/battery packs.

**Comment by romanbetwo** [comment]
Hi guys, "new to the pool" .. So I have growatt SPH10000 . I am running 3 LV batteries in serial- each 10kWh .. Its enough to wake up SPH and DC circuit and discharge/charge almost at 4kW ((52Vx3)*25A). I am currently running Lead Battery settings for SPH but it would be nice to have BMS communication , problem here is that growatt has Low Voltage Cut for lead acid at 12V max .. that means If battery goes down to this level underprotection of battery pops up and BMS (on each battery) will shut the battery down.

Each battery supports mulitple protocols including growatt LV CAN, but SPH 10000 is HV CAN. Can I be the test pilot for this one ? 
Battery POWLAND  has RS485 + CAN + RS232. Batteries have DIP switches config for  "master battery and rest of pack" 

I use Pi4 with 64b os 
Waveshare USB CAN A (shows me cant open /dev/ttyUSB0 - inverter  and /dev/ttyUSB1 - master BMS)

Goal here is (if its possible) sum all battery voltage + SOC and comm with SPH to be able to set SOC discharge and charge limits.
Current lead acid setup handles pretty good all 3 batteries are almost the same %SOC  but sometimes it shut down due to undervoltage protection, as I mentioned LV cut on growatt is 10-12V for Lead acid 

- to explain how this works  
12x 12V lead acid batteries in serie >  144V  /enoughe to "wake" SPH
is the same as 48x 3V Lifepo cells - but has different LVCut 
And also you have to play with max CV (charging voltage) to not exceed 160V .. there is a component on each LV BMS that will insta-short&break to protect the battery  :) 

I mean it survive like 90% of all time, but when consumption is high durring night -  battery goes to 0 - UVP(under voltage protection) you need to manualy power it up.


mail me for more info - bee2twoo at gmail.com

**Comment by ai-republic** [comment]
Ok, let me grasp this:
1. Do you have 12x12V lead acid batteries (144V) or 3 x 10Kwh LiFePo batteries?
2. Those batteries (Lead acid or LiFePo?) are hooked up via BMSes from POWLAND
3. You want to manipulate the BMS data; especially the maximum charging/discharging voltage and SOC
4. And your Growatt SPH10000 is configured to lead acid battery setting instead of Lithium setting?

I hope I got this right?

**Comment by romanbetwo** [comment]
Hi Torsten, I have 3x lifepo powland these batteries has each LV BMS

I connect them in serie to achieve starting voltage of Growatt SPH10000

And yes I want to manupulate charging voltage (current is safe SPH is doing
max 25A) read soc and all necessary data for inverter to work with Lithium
preset.

At this point SPH is configured to lead batt, so no communication is going
on at the moment.

Batteries are connected to each other (i must check the protocol)
I can connect to(Master via DIP switch) battery via rs232 and I can see
each pack stats

Dňa pi 4. 10. 2024, 8:56 Torsten Oltmanns ***@***.***>
napísal(a):

**Comment by ai-republic** [comment]
Thanks for the quick reply. Do you have the CAN communication specification for the POWLAND BMSes? Or can you configure them which specification to use (e.g. Pylon, SMA, Growatt, etc.)?
To realize your project it is possible to use this application. Just need to find out which dialect your BMS speaks 😄 or if I need to implement a new POWLAND BMS binding. The aggregation of the voltage and amps can then be done via a inverter plugin which I can implement very quickly.

**Comment by romanbetwo** [comment]
So Powland has multiple protocols pylon, growatt lv, selpos rest i need to
check

I did check the growatt lv and values  it seems correct

Dňa pi 4. 10. 2024, 9:23 Torsten Oltmanns ***@***.***>
napísal(a):

**Reply by romanbetwo** [reply]
okay let me extend this little bit.

Battery - POWLAND 10kWh
RS485, RS232, CAN
all implemented comm protocols seems to be LV including growatt.
I scanned the battery via CANUSB to see whats going on in battery, i decode the messages and it seems to be OK for growatt Can LV protocol.

I am able to test this program on all kind of products, FOX, SOFAR, Growatt, Huawei, Goodwe, Deye. I work in company that supervise all kind of photovoltic equipment, and I know a lot of people.
Currently at office I have installed growatt MOD10k but its a modbus battery protocol. For SPH its CAN. Growatt sells different type of BMS for each product of these.

**Comment by ai-republic** [comment]
@romanbetwo FYI I just implemented the _GOWATT_MODBUS_ inverter binding.
It would be great if you could also test the other bindings like Growatt, Huawei and Deye. As far as I know the SOFAR is compatible with the Pylon protocol. I'm not sure about the FOX inverters - but I would guess they might also be compatible with the Pylon protocol as most chinese inverters are.

**Reply by romanbetwo** [reply]
Sure, as soon as I get one of them in my hands I'll be able to test and let you know.
